src/components/transactions/Supply/SupplyActions.tsx

```diff
@@ -1,74 +1,240 @@
+    }, [permitAvailable, walletApprovalMethodPreference]);
+
+    const approval = async () => {
+      try {
+        if (requiresApproval && approvedAmount) {
+          if (usePermit) {
+            const deadline = Math.floor(Date.now() / 1000 + 3600).toString();
+            const signatureRequest = await generateSignatureRequest({
+              ...approvedAmount,
+              deadline,
+              amount: parseUnits(amountToSupply, decimals).toString(),
+            });
+
+            const response = await signTxData(signatureRequest);
+            setSignatureParams({ signature: response, deadline, amount: amountToSupply });
+            setApprovalTxState({
+              txHash: MOCK_SIGNED_HASH,
+              loading: false,
+              success: true,
+            });
+          } else {
+            const approveTxData = generateApproval(approvedAmount);
+
+            setApprovalTxState({ ...approvalTxState, loading: true });
+            const response = await sendTx(approveTxData);
+            setApprovalTxState({
+              txHash: response.hash,
+              loading: false,
+              success: true,
+            });
+            fetchApprovedAmount(true);
+          }
+        }
+      } catch (error) {
+        const parsedError = getErrorTextFromError(error, TxAction.GAS_ESTIMATION, false);
+        setTxError(parsedError);
+        setApprovalTxState({
+          txHash: undefined,
+          loading: false,
         });
-      },
-      skip: !amountToSupply || parseFloat(amountToSupply) === 0,
-      deps: [amountToSupply, poolAddress],
-    });
-
-  return (
-    <TxActionsWrapper
-      blocked={blocked}
-      mainTxState={mainTxState}
-      approvalTxState={approvalTxState}
-      isWrongNetwork={isWrongNetwork}
-      requiresAmount
-      amount={amountToSupply}
-      symbol={symbol}
-      preparingTransactions={loadingTxns}
-      actionText={<Trans>Supply {symbol}</Trans>}
-      actionInProgressText={<Trans>Supplying {symbol}</Trans>}
-      handleApproval={() => approval([{ amount: amountToSupply, underlyingAsset: poolAddress }])}
-      handleAction={action}
-      requiresApproval={requiresApproval}
-      tryPermit={usingPermit}
-      sx={sx}
-      {...props}
-    />
-  );
-};
+      }
+    };
+
+    const action = async () => {
+      try {
+        setMainTxState({ ...mainTxState, loading: true });
+        // determine if approval is signature or transaction
+        // checking user preference is not sufficient because permit may be available but the user has an existing approval
+        if (usePermit && signatureParams) {
+          const signedTxData = supplyWithPermit({
+            signature: signatureParams.signature,
+            amount: parseUnits(amountToSupply, decimals).toString(),
+            reserve: poolAddress,
+            deadline: signatureParams.deadline,
+          });
+
+          const response = await sendTx(signedTxData);
+          setMainTxState({
```

Comment:
Need to wait for the tx be confirmed here:

`await response.wait(1);`

Reply:
Added to all `sendTx`'s